pkg/blobstore/badger_blob_access.go

```diff
@@ -0,0 +1,160 @@
+package blobstore
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"log"
+	"time"
+
+	"github.com/buildbarn/bb-storage/pkg/util"
+	"github.com/dgraph-io/badger"
+	duration "github.com/golang/protobuf/ptypes/duration"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	defaultTTL = 10 * 24 * time.Hour
+)
+
+type badgerBlobAccess struct {
+	db            *badger.DB
+	blobKeyFormat util.DigestKeyFormat
+	ttl           time.Duration
+}
+
+// NewBadgerBlobAccess creates a BlobAccess that uses Badger as its backing store.
+func NewBadgerBlobAccess(db *badger.DB, blobKeyFormat util.DigestKeyFormat, ttl *duration.Duration) BlobAccess {
+	ba := &badgerBlobAccess{
+		db:            db,
+		blobKeyFormat: blobKeyFormat,
+		ttl:           defaultTTL,
+	}
+	if ttl != nil {
+		ba.ttl = time.Duration(ttl.Seconds) * time.Second
+	}
+	go ba.gc()
+	return ba
+}
+
+// Badger storage garbage collection
+func (ba *badgerBlobAccess) gc() {
+	ticker := time.NewTicker(5 * time.Minute)
```

Comment:
Out of curiosity, what is the reasoning behind choosing those values for garbage collection? Should they be configurable as well for end user?